tools/bpf2c/bpf_code_generator.cpp

```diff
@@ -557,17 +548,12 @@ bpf_code_generator::encode_instructions(const bpf_code_generator::unsafe_string&
                 output.lines.push_back(format_string("%s = -(int64_t)%s;", destination, destination));
                 break;
             case AluOperations::Mod:
-                if (inst.opcode & EBPF_SRC_REG) {
-                    output.lines.push_back(check_div_by_zero);
-                } else if (inst.imm == 0) {
-                    throw bpf_code_generator_exception(
-                        "invalid instruction - constant division by zero", output.instruction_offset);
-                }
+                output.lines.push_back(format_string("if (%s != 0) {", source));
                 if (is64bit)
-                    output.lines.push_back(format_string("%s %%= %s;", destination, source));
+                    output.lines.push_back(format_string(INDENT "%s %%= %s;\n" INDENT "}", destination, source));
                 else
-                    output.lines.push_back(
-                        format_string("%s = (uint32_t)%s %% (uint32_t)%s;", destination, destination, source));
+                    output.lines.push_back(format_string(
+                        INDENT "%s = (uint32_t)%s %% (uint32_t)%s;\n" INDENT "}", destination, destination, source));
                 break;
```

Comment:
Where is the else case where source == 0?
I see the else clause in 525-530 for division, but I don't see the equivalent here for modulo.